marshal/tests/serialization/pointers.go

```diff
@@ -0,0 +1,54 @@
+package serialization
+
+import (
+	"errors"
+	"reflect"
+)
+
+// ErrFirstPtrChanged this error indicates that a double or single reference was passed to the unmarshal function
+// (example (**int)(**0) or (*int)(*0)) and the function did overwrite the first reference.
+var ErrFirstPtrChanged = errors.New("unmarshal function rewrote first pointer")
+
+// ErrSecondPtrNotChanged this error indicates that a double value reference was passed to the unmarshal function
+// (example (**int)(**0)) and the function did not overwrite the second reference.
+// Of course, it's not friendly to the garbage collector, overwriting references to values all the time,
+// but this is the current realization `gocql` and changing it can lead to unexpected results, because unmarshal functions are used in loops.
```

Comment:
```suggestion
// ErrSecondPtrNotChanged this error indicates that a double reference was passed to the Unmarshal function
// (example (**int)(**0)) and the function did not overwrite the second reference.
// Of course, it's not friendly to the garbage collector, overwriting references to values all the time,
// but this is the current implementation `gocql` and changing it can lead to unexpected results in some cases.
```